test/utils/create-repositories.ts

```diff
@@ -0,0 +1,30 @@
+import { Subscription } from "models/subscription";
+import { RepoSyncState } from "models/reposyncstate";
+
+const generateUniqueRepoOwner = (): string => {
+	const prefix = "repo-owner";
+	const uniqueId = Math.floor(Math.random() * 1000); // Generate a random number
+
+	return `${prefix}-${uniqueId}`;
+};
+
+export const createRepositories = async (subscriptions: Subscription[]): Promise<RepoSyncState[]> => {
```

Comment:
/nit
Up for discussion, I personally find sharing data across multiple tests is not very flexible in long term. I'd prefer "creating the prepare data" right within the test, and even better, right within the test function `it`. 
One big problem I have with sharing data is whenever I try to change something in the data, I may broken another test even though I am not touching that bit. So the data become very carefully constructed and a bit fragile across the tests. Are these functions expected to be used by other tests cases?

ps: @bgvozdev previously created database-state-creator.ts to create some subscriptions/repos, not sure those can/should be used in your above test cases.

Reply:
Good call out. Will move them.

Reply:
Moved them into the test file. Had a look at what we already have in database-state-creator.ts but it doesn't really give me the results I want so sticking with these util functions.